src/ln/channelmanager.rs

```diff
@@ -1145,9 +1145,18 @@ impl ChainListener for ChannelManager {
 				if let Ok(Some(funding_locked)) = chan_res {
 					let announcement_sigs = match self.get_announcement_sigs(channel) {
 						Ok(res) => res,
-						Err(_e) => {
-							//TODO: push e on events and blow up the channel (it has bad keys)
-							return true;
+						Err(e) => {
+							if e.action.is_some() {
+								new_events.push(events::Event::DisconnectPeer {
+									node_id: channel.get_their_node_id(),
+									msg: None,
+								});
+								failed_channels.push(channel.force_shutdown());
+								short_to_id.remove(&channel.get_short_channel_id().unwrap());
+								return false;
+							} else {
+								return true;
```

Comment:
What to do with failure due to not having received FundingLocked yet from remote peer ? Do we need to scan the block again ? And so flagged it somewhere ? Error case from get_channel_announcement.